src/Npgsql/NpgsqlConnection.cs

```diff
@@ -180,14 +180,10 @@ public override Task OpenAsync(CancellationToken cancellationToken)
         void GetPoolAndSettings()
         {
             var pools = PoolManager.Pools;
-            lock (pools)
+            if (pools.TryGetValue(_connectionString, out _pool))
+                Settings = _pool.Settings;  // Great, we already have a pool
```

Comment:
Can you please return from the method after setting `Settings`? This would obviate the else.